### PR TITLE
feat: show most recent contributions for contributors instead of last 30 days

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Updates contributor pull requests hooks to conditionally show last 30 PRs regardless of time instead of last 30 days of PRs
- Needs https://github.com/open-sauced/api/pull/407

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/engineering/issues/194

## Mobile & Desktop Screenshots/Recordings

BEFORE:

image

AFTER:

image

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
Pending the pizza-time update, we should eventually see if we can make these charts commit (count) based instead of PR (count) based. Something we will need to explore design wise as well. 

cc @jpmcb @isabensusan

File: lib/hooks/useContributorPullRequestsChart.ts [hunk 1]
```diff
@@ -5,7 +5,13 @@ import { getAvatarByUsername } from "lib/utils/github";
 import getFormattedTooltipValue from "lib/utils/get-formatted-tooltip-value";
 import useContributorPullRequests from "./api/useContributorPullRequests";
 
-const useContributorPullRequestsChart = (contributor: string, topic: string, repoIds: number[] = [], range = 30) => {
+const useContributorPullRequestsChart = (
```
Comment (nickytonline) [inline lib/hooks/useContributorPullRequestsChart.ts:8]: Refeactor (non-blocking): There's quite a few parameters now. Consider promoting them to one option parameter with all the current parameters.

File: lib/hooks/api/useContributorPullRequests.ts [hunk 2]
```diff
@@ -15,7 +15,8 @@ export function getContributorPRUrl(
   topic: string,
   repoIds: number[] = [],
   limit = 8,
-  range = 30
+  range = 30,
+  mostRecent = false
```
Comment (nickytonline) [inline lib/hooks/api/useContributorPullRequests.ts:19]: Refeactor (non-blocking): There's quite a few parameters now. Consider promoting them to one option parameter with all the current parameters.

File: lib/hooks/api/useContributorPullRequests.ts [hunk 3]
```diff
@@ -32,6 +33,10 @@ export function getContributorPRUrl(
     query.set("repoIds", repoIds.join(","));
   }
 
+  if (mostRecent) {
+    query.set("rangeType", "all");
+  }
+
   query.set("range", `${range}`);
```
Comment (nickytonline) [inline lib/hooks/api/useContributorPullRequests.ts:40]: Is `range` still required if we're saying get all via

```javascript
  if (mostRecent) {
    query.set("rangeType", "all");
  }
```
?

Comment (brandonroberts) [reply lib/hooks/api/useContributorPullRequests.ts]: Good catch, fixed

**Review by nickytonline — APPROVED** [review]
Great stuff! 🚢

**Review by NsdHSO — APPROVED** [review]

**Review by bdougie — APPROVED** [review]
Let's go!

**Review by OgDev-01 — APPROVED** [review]

**Pull request merged**